touch and display example for STM32F469I-DISCO B08 board revision (NT35510 LCD controller)

Amperstrand opened the pull request:
## Summary

This PR adds NT35510 LCD controller support for the STM32F469I-DISCO B08 board revision, enabling both display and touch functionality.

I'm not much of a rust person, and my main goal was just trying to get it to work (which it now does). The code has been restructured into what I hope are clean, reusable commits with proper attribution to the original sources.

## What's Included

**Commits:**
1. `feat(ltdc)`: DSI constructor, framebuffer DrawTarget, and layer management APIs
2. `feat(display)`: NT35510 driver and F469DISCO board init helpers
3. `feat(examples)`: F469DISCO display examples with runtime panel autodetection
4. `fix(examples)`: Improved lcd-test panel detection and init sequence

**Attribution:**
- NT35510 init sequence based on [diybitcoinhardware/f469-disco](https://github.com/diybitcoinhardware/f469-disco) (MIT license)
- Display timing and patterns from STMicroelectronics STM32CubeF4 BSP (BSD-3-Clause)

## Hardware Tested

- ✅ STM32F469I-DISCO B08 revision (NT35510 LCD panel)
- ✅ Display initialization and test patterns
- ✅ Touch controller (FT6X06) calibration and input

## Questions

NT35510 could possibly be its own crate like otm8009a is, or alternatively it could live in `src/display/` as I've structured it here. Open to feedback on the best approach.

tegimeki reviewed (commented):
Glad to see support for the newer display, and have some comments on compatibility with the older one.  A more modular BSP-level display support approach might resolve these issues, allowing code to first probe the display type with a minimal DSI host config and then initialize for the specific display from there.

Copilot committed:
Initial plan

Copilot committed:
Add multi-controller support and touchscreen integration to f469disco-lcd-test

Co-authored-by: Amperstrand <141745238+Amperstrand@users.noreply.github.com>

Copilot committed:
Address code review feedback: rename pattern_timer, extract error throttle helper

Co-authored-by: Amperstrand <141745238+Amperstrand@users.noreply.github.com>

Copilot committed:
Final review improvements: add double-init guard, clarify comments, extract I2C address constant

Co-authored-by: Amperstrand <141745238+Amperstrand@users.noreply.github.com>

Amperstrand committed:
Merge pull request #6 from Amperstrand/copilot/compare-screen-touch-implementations

Port multi-controller support and touchscreen integration from reference implementation

tegimeki commented:
I tried your latest on a B01 board with the OTM8009A display:  it looks good with auto-detect but when running with `otm8009a-only` I see a black stripe down one side of the display.  I haven't looked closely yet but can do that, unless you already know/expect the behavior there.

Also, I get `detect_touch timed out after 3 attempts` output continuously and can double-check that my touch controller is still working (it was last I tried a while back).

burrbull commented on Cargo.toml:
```diff
@@ -532,6 +532,8 @@ gpiok = []
 i2c3 = []
 lptim1 = []
 ltdc = ["dep:micromath"]
+nt35510-only = []
```
I'm not happy with example related features here.

Maybe we should restructure `examples` directory to have its own `Cargo.toml`.
Integrate `xtask` maybe like in github.com/esp-rs/esp-hal/ ?

Amperstrand replied:
Apologies for spamming the pull request with AI generated content. At this point I am mostly experimenting and probably won't get back to it in the near future. NT35510 doesn't have its own crate like otm8009a does, but maybe it should.

My main goal was really mostly just to get the stm32f469 going with all revisions in a way where it's easy to just get it working

burrbull replied:
You could convert the PR into draft until it is complete.

Copilot committed:
Initial plan

Copilot committed:
Fix rustfmt issues and add changelog entry

Co-authored-by: Amperstrand <141745238+Amperstrand@users.noreply.github.com>

Amperstrand committed:
Merge pull request #16 from Amperstrand/copilot/fix-formatting-errors

Resolve rustfmt/changelog CI failures in STM32F469 Disco display-touch updates

Copilot committed:
Initial plan

Copilot committed:
fix(display): update OTM8009A timing to ST official values

Co-authored-by: Amperstrand <141745238+Amperstrand@users.noreply.github.com>

Amperstrand committed:
Merge pull request #17 from Amperstrand/copilot/update-otm8009a-timing

fix(display): align OTM8009A LTDC timing with ST reference values on F469-DISCO

Commit:
feat(ltdc): add DSI constructor, framebuffer DrawTarget, and layer management APIs

Commit:
feat(display): add NT35510 driver and F469DISCO board init helpers

Commit:
feat(examples): add F469DISCO display examples with runtime panel autodetection

Commit:
fix(examples): improve lcd-test panel detection and init sequence

- Add 20ms delay after DSI bus turnaround for panel link settle
- Add 20ms settle delay in detect_lcd_controller before probing
- Move I2C init before LCD reset (proper hardware sequencing)
- Fix smart fallback: default to NT35510 when probe inconclusive
  (previously incorrectly fell back to OTM8009A on B08 boards)

Tested on STM32F469I-DISCO B08 (NT35510 panel) - display now works.

Commit:
fix(examples): remove animated-layers, add touch calibration

- Remove f469disco-animated-layers.rs (was showing green screen due to
  transparent layer config issue)
- Remove from Cargo.toml
- Add f469disco-touch-debug.rs for interactive touch calibration
- Add TOUCH_X_OFFSET / TOUCH_Y_OFFSET constants to paint.rs
  for future calibration adjustments

Amperstrand commented:
Closing in favor of two cleaner, split PRs: one for core DSI/LTDC improvements and one for F469-Disco examples. The NT35510 driver has been extracted to a standalone crate (https://crates.io/crates/nt35510).
